src/backend/cpu/copy.cpp

```diff
@@ -63,6 +63,30 @@ namespace cpu
         return out;
     }
 
+    template<typename T>
+    void multiply_inplace(Array<T> &in, double val)
```

Comment:
Can't we do this using `detail::copyArray<inType, outType>` if we modify it to have scale parameter ? like the following
```
template<typename inType, typename outType>
void copyArray(Array<outType> &out, Array<inType> const &in, double scale_factor);
//....
Array<T> input = ....;
copyArray(input, input, scale_value);
```
Just wondering if that is possible.